src/layer_4/tcp.rs

```diff
@@ -105,7 +105,7 @@ pub fn repl<'a, 'b>(
             tcp_repl.set_sequence(tcp_req.get_acknowledgement());
         }
         /* Answer to SYN */
-        flags if flags & TcpFlags::SYN == TcpFlags::SYN => {
+        flags if flags == TcpFlags::SYN => {
```

Comment:
I'm not sure to agree here: it is possible to have other flags set (SYN+ECN+CWR for example).

In firewall rules we often specify `flags S/SA` which means "from syn and ack flags, only syn is set". Maybe we could do the same here?

```suggestion
        flags if flags & (TcpFlags::SYN | TcpFlags::ACK) == TcpFlags::SYN => {
```

Reply:
What about packets with other flags that would be inconsistent, as for example `SYN` + `FIN`?

Reply:
Good point. No idea. Should we mimic "classical" networking stacks?